plugins/network/NetClass.hpp

```diff
@@ -1,38 +1,86 @@
 ﻿#ifndef __NETCLASS_HPP__
 #define __NETCLASS_HPP__
 
+#include <memory>
+#include <string>
+#include <list>
 #include <plugin.hpp>
+
 #ifdef NETWORK_LOGGING
-#include <stdio.h>
+#include <cstdio>
 #endif
 
-class NetResourceList
+using string = std::wstring;
+
+class NetResource
 {
-private:
-	NETRESOURCE* ResList;
-	unsigned int ResCount;
+public:
+	DWORD dwScope{};
+	DWORD dwType{};
+	DWORD dwDisplayType{};
+	DWORD dwUsage{};
+	string lpLocalName{};
+	string lpRemoteName{};
+	string lpComment{};
+	string lpProvider{};
 
 public:
-	NetResourceList();
-	~NetResourceList();
+	NetResource() = default;
+
+	NetResource(const NetResource& other) = default;
+	NetResource& operator=(const NetResource& other) = default;
+	NetResource(NetResource&& other) noexcept = default;
+	NetResource& operator=(NetResource&& other) noexcept = default;
+
+	explicit NetResource(NETRESOURCE& other)
+	{
+		dwScope = other.dwScope;
+		dwType = other.dwType;
+		dwDisplayType = other.dwDisplayType;
+		dwUsage = other.dwUsage;
+		lpLocalName = other.lpLocalName? other.lpLocalName : L"";
+		lpRemoteName = other.lpRemoteName? other.lpRemoteName : L"";
+		lpComment = other.lpComment? other.lpComment : L"";
+		lpProvider = other.lpProvider? other.lpProvider : L"";
+	}
+
+	static NETRESOURCE getNETRESOURCE(const NetResource* resource)
```

Comment:
Why a static function? It needs a class instance anyway.

Reply:
in NetResourceList::Enumerate  NetResource maybe is null . In this case getNETRESOURCE return empty NETRESOURCE

Reply:
It's kinda fishy that we then pass that we then pass that empty NETRESOURCE to [WNetOpenEnum](https://docs.microsoft.com/en-us/windows/win32/api/winnetwk/nf-winnetwk-wnetopenenumw), which expects either a null pointer or a properly filled structure.

In the current version we pass the ptr as is, without turning nullptr to an empty struct:
https://github.com/FarGroup/FarManager/blob/f20c908cfe7746c4f9d425c3548f18a3b1b82a42/plugins/network/NetClass.cpp#L154

Are you sure you want an empty struct there?

Reply:
fixed